lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1111,6 +1111,12 @@ LLVMToSPIRVBase::getLoopControl(const BranchInst *Branch,
 
   size_t LoopControl = spv::LoopControlMaskNone;
   std::vector<std::pair<SPIRVWord, SPIRVWord>> ParametersToSort;
+  // We need to create these variables here because it might be the case when
+  // metadata is not defined for one or two of these parameters and in this
+  // case we need to store "-1" value
+  int64_t LoopCountMin = -1;
+  int64_t LoopCountMax = -1;
+  int64_t LoopCountAvg = -1;
```

Comment:
An alternative is to wrap these numbers in a function-scope struct with a name that would correspond to the implemented extension, something akin to:
```
struct LoopCountInfo {
  int64_t Min = -1, Max = -1, Avg = -1;
} LoopCount;
```
Not sure if it gains much in terms of code readability (the function still has to be decluttered some day), but one could argue that a single struct-typed object is easier to eye-track than 3 distinct variables.

Reply:
```
struct LoopCountInfo {
  static constexpr int64_t Min = -1;
  static constexpr Max = -1;
  static constexpr Avg = -1;
};
```

Ideally, such struct (or just a bunch of constexpr variables) should reside in `spv` or `spv::internal` namespace similar to `spv::LoopControlMaskNone`. The idea is that if we have some special value for this mask, let's define it somewhere near the mask definition.

Reply:
The only thing I'd argue with here is `static constexpr` - different loops may want to use different instances of the struct, and the value of the fields may change based on the actual attribute values.

Reply:
My understanding was that here we are talking about some generic default values, which probably should be documented in the corresponding SPIR-V extension and even if they depend on the attribute, I would assume that there are finite set of such default values for each attribute, which means that we can still define them as `constexpr`

Reply:
> My understanding was that here we are talking about some generic default values

Not quite; these are needed as in-scope variables that are assigned to on lines 1219-1233 and read from on lines 1238-1253.